http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GKEInterpreter.scala

```diff
@@ -0,0 +1,390 @@
+
+        cidrBuilder = CidrBlock.newBuilder()
+        _ <- F.delay(config.clusterConfig.authorizedNetworks.foreach(cidrIP => cidrBuilder.setCidrBlock(cidrIP.value)))
+
+        //TODO comb through with security reqs in mind
+        createClusterReq = Cluster
+          .newBuilder()
+          .setName(dbCluster.clusterName.value)
+          .addNodePools(
+            getGoogleNodepool(defaultNodepool)
+          )
+          // all the below code corresponds to security recommendations
+          .setLegacyAbac(LegacyAbac.newBuilder().setEnabled(false))
+          .setNetwork(kubeNetwork)
+          .setSubnetwork(kubeSubNetwork)
+          .setNetworkPolicy(
+            NetworkPolicy
+              .newBuilder()
+              .setEnabled(true)
+          )
+          .setMasterAuthorizedNetworksConfig(
+            MasterAuthorizedNetworksConfig
+              .newBuilder()
+              .setEnabled(true)
+              .addAllCidrBlocks(
+                config.clusterConfig.authorizedNetworks
+                  .map(ip => CidrBlock.newBuilder().setCidrBlock(ip.value).build())
+                  .asJava
+              )
+          )
+          .setIpAllocationPolicy( //otherwise it uses the legacy one, which is insecure. See https://cloud.google.com/kubernetes-engine/docs/how-to/alias-ips
+            IPAllocationPolicy
+              .newBuilder()
+              .setUseIpAliases(true)
+          )
+          .build()
+
+        req = KubernetesCreateClusterRequest(dbCluster.googleProject, dbCluster.location, createClusterReq)
+        op <- gkeService.createCluster(req)
+        lastOp <- gkeService
+        //TODO: refactor this to handle interupts
+          .pollOperation(
+            KubernetesOperationId(dbCluster.googleProject, dbCluster.location, op),
+            config.monitorConfig.clusterCreate.interval,
+            config.monitorConfig.clusterCreate.maxAttempts
+          )
+          .compile
+          .lastOrError
+        _ <- if (lastOp.isDone) logger.info(s"Create cluster operation has finished for cluster ${dbCluster.id}")
+        else
+          logger.error(s"Create cluster operation has failed for cluster ${dbCluster.id}") >> F.raiseError[Unit](
+            ClusterCreationException(
+              s"Failed to poll cluster creation operation to completion for cluster ${dbCluster.id} and default nodepool ${defaultNodepool.id}"
+            )
+          )
+        getClusterOpt <- gkeService.getCluster(dbCluster.getGkeClusterId)
+        getCluster <- F.fromOption(
+          getClusterOpt,
+          ClusterCreationException(
+            s"Failed to get cluster to retrieve external IP after successful creation polling for ${dbCluster.id}"
+          )
+        )
+        _ <- kubernetesClusterQuery
+          .updateAsyncFields(dbCluster.id,
+                             KubernetesClusterAsyncFields(
+                               IP(getCluster.getEndpoint()),
```

Comment:
Hm what does `getEndpoint()` give us? Not sure it's the right IP -- we really need the external IP of the nginx controller's LoadBalancer (which isn't in yet because it needs to be helm installed).

Reply:
Oh, I didn't know it wasn't the cluster IP anymore. I'll replace this with a TODO then, it seems we need to wait on the helm command to get this.